tests/unit/amo/components/TestSectionLinks.js

```diff
@@ -64,4 +72,67 @@ describe('SectionLinks Component', () => {
     expect(root.find('.SectionLinks-link--active').children())
       .toIncludeText('Themes');
   });
+
+  it('renders Language Tools active when viewContext is languageTools', () => {
+    _store.dispatch(setViewContext(VIEW_CONTEXT_LANGUAGE_TOOLS));
+    const root = render();
+
+    expect(root.find('.SectionLinks-dropdownlink--active').children())
+      .toIncludeText('Dictionaries & Language Packs');
+  });
+
+  it('shows Firefox name and hides link in header on Desktop', () => {
+    _store.dispatch(setClientApp(CLIENT_APP_FIREFOX));
+    const root = render();
+
+    expect(root.find('.SectionLinks-subheader').at(0).children())
+      .toIncludeText('for Firefox');
+    expect(root.find(`.SectionLinks-clientApp-${CLIENT_APP_ANDROID}`))
+      .toHaveLength(1);
+    expect(root.find(`.SectionLinks-clientApp-${CLIENT_APP_FIREFOX}`))
+      .toHaveLength(0);
+  });
+
+  it('shows Android name and hides link in header on Android', () => {
+    _store.dispatch(setClientApp(CLIENT_APP_ANDROID));
+    const root = render();
+
+    expect(root.find('.SectionLinks-subheader').at(0).children())
+      .toIncludeText('for Firefox for Android');
+    expect(root.find(`.SectionLinks-clientApp-${CLIENT_APP_ANDROID}`))
+      .toHaveLength(0);
+    expect(root.find(`.SectionLinks-clientApp-${CLIENT_APP_FIREFOX}`))
+      .toHaveLength(1);
+  });
+
+  it('changes clientApp when different site link clicked', () => {
+    const dispatchSpy = sinon.spy(_store, 'dispatch');
+    const fakeEvent = createFakeEvent({
+      currentTarget: {
+        getAttribute: (attribute) => {
+          if (attribute === 'data-clientapp') {
+            return CLIENT_APP_ANDROID;
+          }
+          if (attribute === 'href') {
+            return `/en-US/${CLIENT_APP_ANDROID}/`;
+          }
+
+          return undefined;
+        },
+      },
+    });
+    const getAttributeSpy = sinon.spy(fakeEvent.currentTarget, 'getAttribute');
+    const fakeRouter = { push: sinon.stub() };
+    _store.dispatch(setClientApp(CLIENT_APP_FIREFOX));
+    const root = render({ router: fakeRouter });
+
+    root.find(`.SectionLinks-clientApp-android`)
+      .simulate('click', fakeEvent);
+
+    sinon.assert.called(fakeEvent.preventDefault);
+    sinon.assert.calledWith(getAttributeSpy, 'data-clientapp');
+    sinon.assert.calledWith(getAttributeSpy, 'href');
+    sinon.assert.calledWith(dispatchSpy, setClientApp(CLIENT_APP_ANDROID));
+    sinon.assert.calledWith(fakeRouter.push, `/en-US/${CLIENT_APP_ANDROID}/`);
+  });
```

Comment:
I would have expected a test to make sure we render a `DropdownMenu`.

Reply:
Oh, good call. Will add one.